### ISSUE TITLE
Plugins not loaded from `~/.prototools`?

### ISSUE THREAD

**Issue by rotu** [moonrepo/proto#273]
### What version?

0.21.1

### What happened?

In trying to install moon off the instructions [here](https://moonrepo.dev/docs/install), I get an error:

```
proto add-plugin moon "source:https://raw.githubusercontent.com/moonrepo/moon/master/proto-plugin.toml"; proto install moon
[ INFO 2023-11-06 15:03:48] proto::commands::add_plugin:add_plugin  Added plugin moon to local ~\.prototools
Error: proto::tool::unknown

  × moon is not a built-in tool or has not been configured as a plugin, unable to proceed.
```

I think this is related to the fact that I'm running this from my home directory. It doesn't seem to happen when adding the plugin globally or when adding it locally in a subdirectory of my user home.

Note that in the below trace, I *don't* see a line like "proto_core::tools_config:install  Loading .prototools"

### Any logs?

```
proto install moon --log trace
[DEBUG 2023-11-06 15:08:42] proto  Running proto v0.21.1  args=["C:\\Users\\dan\\.proto\\bin\\proto.exe", "install", "moon", "--log", "trace"]
[TRACE 15:08:42] starbase::app  Running startup phase with 1 systems
[TRACE 15:08:42] starbase::app  Running analyze phase with 1 systems
[TRACE 15:08:42] starbase::app  Running execute phase with 2 systems
[DEBUG 15:08:42] proto_core::user_config:install  Loading config.toml  file="C:\\Users\\dan\\.proto\\config.toml"
[TRACE 15:08:42] starbase_utils::fs:install  Reading file  file="C:\\Users\\dan\\.proto\\config.toml"
[DEBUG 15:08:42] proto_core::tool_loader:install  Traversing upwards to find a configured plugin  tool="moon"
[TRACE 15:08:42] starbase::app  Running shutdown phase with 1 systems
Error: proto::tool::unknown

  × moon is not a built-in tool or has not been configured as a plugin, unable to proceed.
```

### Operating system?

Windows

### Architecture?

x64

**Comment by milesj** [comment]
We don't traverse passed the home dir: https://github.com/moonrepo/proto/blob/master/crates/core/src/tool_loader.rs#L132

So it's just short-circuiting earlier. We can move that line below the logic that finds the tool.

**Comment by rotu** [comment]
Makes sense. Maybe `proto add-plugin` should also throw up a warning/error too when it's in a parent directory of `$HOME`.

**Comment by alf5** [comment]
i am also getting this same error. im running in a kubuntu environment.
im executing the commands in the installation docs for a proto install while in my $HOME directory.

**Comment by milesj** [comment]
Fixed on 0.22
